### PR TITLE
Pinning Numpy versions

### PR DESCRIPTION
**Pull request by aspctu**
Because `numpy v1.24.0` breaks `np.float` usage which `table_logger` (which is used in `kfserving`) utilizes, this PR fixes the breaking Numpy changes.

### CODE REVIEW

**Comment by pankajroark** [comment]
This can still fail when requirements are inferred. I have a fix for it in another PR. Perhaps we can take this specific fox from there, there are lots of other changes in that PR.

**Comment by aspctu** [comment]
@pankajroark ack, updated the pr. i placed it in the `base.Dockerfile.jinja` that we currently have

**Comment by pankajroark** [comment]
> you're going to have to make a new truss release by updating the poetry package version. I might be worth just pinning this in pankaj's PR

Poetry publish isn't critical, but we'll need to push the truss image to dockerhub and use that in Baseten, and deploy Baseten.

File: truss/templates/base.Dockerfile.jinja [hunk 1]
```diff
@@ -83,6 +83,16 @@ WORKDIR $APP_HOME
 {% endblock %}
 
 
+# This is a hack, kfserving uses table_logger, which doesn't work well with
+# numpy 1.24 onwards, where np.float and np.int have been remove.
+# https://github.com/AleksTk/table-logger/blob/v0.3.6/table_logger/table_logger.py#L80
+# Monkey patch table_logger here. Ultimately we should move away from kfserving,
+# perhaps to kserve.
```
Comment (howespt) [inline truss/templates/base.Dockerfile.jinja:90]: @zero1zero is working on this

**Pull request merged**